airbyte-cdk/python/unit_tests/sources/streams/http/test_http_client.py

```diff
@@ -391,3 +401,57 @@ def update_response(*args, **kwargs):
 
     assert http_client._session.send.call_count == call_count
     assert returned_response == valid_response
+
+
+def test_disable_retries():
```

Comment:
These tests assert on private parameters. If we change the internal logic without changing the behavior, those will break. Should we instead do a test with requests_mock that ensure that we perform only one query in case of failure? The same question applies to the tests below

Reply:
Mocked